SignalMessaging/attachments/ApproveContactShareViewController.swift

```diff
@@ -0,0 +1,556 @@
+    }
+
+    override public func loadView() {
+        super.loadView()
+
+        self.navigationItem.title = NSLocalizedString("CONTACT_SHARE_APPROVAL_VIEW_TITLE",
+                                                      comment: "Title for the 'Approve contact share' view.")
+
+        self.view.preservesSuperviewLayoutMargins = false
+        self.view.backgroundColor = UIColor.white
+
+        updateContent()
+
+        updateNavigationBar()
+    }
+
+    // TODO: Surface error with resolution to user if not.
+    func canShareContact() -> Bool {
+        return contactShare.ows_isValid()
+    }
+
+    func updateNavigationBar() {
+        self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel,
+                                                                target: self,
+                                                                action: #selector(didPressCancel))
+
+        if canShareContact() {
+            self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: NSLocalizedString("ATTACHMENT_APPROVAL_SEND_BUTTON",
+                                                                                              comment: "Label for 'send' button in the 'attachment approval' dialog."),
+                                                                     style: .plain, target: self, action: #selector(didPressSendButton))
+        } else {
+            self.navigationItem.rightBarButtonItem = nil
+        }
+
+    }
+
+    private func updateContent() {
+        SwiftAssertIsOnMainThread(#function)
+
+        guard let rootView = self.view else {
+            owsFail("\(logTag) missing root view.")
+            return
+        }
+
+        for subview in rootView.subviews {
+            subview.removeFromSuperview()
+        }
+
+        let scrollView = UIScrollView()
+        scrollView.preservesSuperviewLayoutMargins = false
+        self.view.addSubview(scrollView)
+        scrollView.layoutMargins = .zero
+        scrollView.autoPinWidthToSuperview()
+        scrollView.autoPin(toTopLayoutGuideOf: self, withInset: 0)
+        scrollView.autoPinEdge(toSuperviewEdge: .bottom)
+
+        let fieldsView = createFieldsView()
+
+        scrollView.addSubview(fieldsView)
+        fieldsView.autoPinLeadingToSuperviewMargin()
+        fieldsView.autoPinTrailingToSuperviewMargin()
+        fieldsView.autoPinEdge(toSuperviewEdge: .top)
+        fieldsView.autoPinEdge(toSuperviewEdge: .bottom)
+    }
+
+    private func createFieldsView() -> UIView {
+        SwiftAssertIsOnMainThread(#function)
+
+        var rows = [UIView]()
+
+        rows.append(createNameRow())
+
+        for fieldView in fieldViews {
+            rows.append(fieldView)
+        }
+
+        return ContactFieldView(rows: rows, hMargin: hMargin)
+    }
+
+    private let hMargin = CGFloat(16)
+
+    func createNameRow() -> UIView {
+        let nameVMargin = CGFloat(16)
+
+        let row = UIView()
```

Comment:
Ah this is similar to what we were talking about yesterday - why this wrapper view?

You can set the margins right on the stack view (but remember to set `stackView.isLayoutMarginsRelativeArrangement = true`)

Reply:
This was just a rough MVP to unblock testing.  I've removed this in a subsequent PR.